collections.go

```diff
@@ -125,23 +127,49 @@ func (cl *CollectionsListClient) Do(ctx context.Context, fn CollectionsListFn) e
 	}
 }
 
-type CollectionsCreateClient struct {
-	sl *sling.Sling
+// collectionsCreateParams represents the Outline Collections.create parameters
+type collectionsCreateParams struct {
+	Name        string     `json:"name"`
+	Description string     `json:"description,omitempty"`
+	Permission  Permission `json:"permission,omitempty"`
+	Color       string     `json:"color,omitempty"`
+	Private     bool       `json:"private,omitempty"`
 }
 
-func newCollectionsCreateClient(sl *sling.Sling, name string) *CollectionsCreateClient {
-	data := struct {
-		Name string `json:"name"`
-	}{Name: name}
+type CollectionsCreateClient struct {
+	sl     *sling.Sling
+	params collectionsCreateParams
+}
 
+func newCollectionsCreateClient(sl *sling.Sling, params collectionsCreateParams) *CollectionsCreateClient {
 	copy := sl.New()
-	copy.Post(common.CollectionsCreateEndpoint()).BodyJSON(&data)
+	return &CollectionsCreateClient{sl: copy, params: params}
+}
 
-	return &CollectionsCreateClient{sl: copy}
+func (cl *CollectionsCreateClient) Description(desc string) *CollectionsCreateClient {
+	cl.params.Description = desc
+	return cl
+}
+
+func (cl *CollectionsCreateClient) Permission(perm Permission) *CollectionsCreateClient {
+	cl.params.Permission = perm
+	return cl
+}
```

Comment:
Better to have: `PermissionRead` and `PermissionReadWrite` methods so that user cannot give random values at all: make API harder to misuse. If you agree then `Permission` type is not needed.

Reply:
Checkout https://github.com/ioki-mobility/go-outline/pull/29/commits/5487012e8eb292419e0d3d79ec772965033621d3